js/client.js

```diff
@@ -1011,6 +1021,72 @@
 				break;
 			}
 		},
+		parseGroups: function (groupsList) {
+			var data = null;
+			try {
+				data = JSON.parse(groupsList);
+			} catch (e) {}
+			if (!data) return; // broken JSON - keep default ranks
+
+			var authTiers = [[], [], []]; // we will be splitting them into three tiers, before compiling them back into the main tree
```

Comment:
That should probably be handled server-side? The server should send them in the correct order.